monai/handlers/checkpoint_loader.py

```diff
@@ -92,15 +105,22 @@ def __call__(self, engine: Engine) -> None:
         """
         checkpoint = torch.load(self.load_path, map_location=self.map_location)
 
-        if not self.strict_shape:
-            k, _ = list(self.load_dict.items())[0]
-            # single object and checkpoint is directly a state_dict
-            if len(self.load_dict) == 1 and k not in checkpoint:
-                checkpoint = {k: checkpoint}
+        k, _ = list(self.load_dict.items())[0]
+        # single object and checkpoint is directly a state_dict
+        if len(self.load_dict) == 1 and k not in checkpoint:
+            checkpoint = {k: checkpoint}
 
-            # skip items that don't match data shape
+        if not self.strict_shape:
+            pop_items: List[str] = []
             for k, obj in self.load_dict.items():
-                checkpoint[k] = copy_model_state(obj, checkpoint, inplace=False)[0]
+                if isinstance(obj, (torch.nn.Module)):
+                    # skip items that don't match key name or data shape
+                    checkpoint[k] = copy_model_state(obj, checkpoint, inplace=False)[0]
```

Comment:
@Nic-Ma according to the docstring of `copy_model_state`, the values of `obj` will be overwritten by the ones from `checkpoint` when their keys match (also see [here](https://github.com/Project-MONAI/MONAI/blob/ce45640b2aab6e094e3c8af9ea49634497df9fb6/monai/networks/utils.py#L385-L386)). So for example, if the current model is `nn.Linear(4096, 10)`, and the corresponding Tensor from checkpoint has shape `(4096, 1000)`, the returned `state_dict` will have a tensor with shape `(4096, 1000)` right? Is this the desirable behavior? In most cases, we hope to skip loading the unmatched weights rather than let them overwrite the current model's weights.

Reply:
that's the expected behaviour because the user explicitly provides a `mapping` dict here for the model keys